### PR TITLE
add docments to 32_text.models.awdlstm notebook

### LINKED ISSUE #22: 32_text.models.awdlstm: create docments
Add documents to [32_text.models.awdlstm](https://github.com/muellerzr/fastai-docment-sprint/blob/master/nbs/32_text.models.awdlstm.ipynb).

See the style guide and contribution guide for more details (links will be added when created).

This requires a moderate familiarity with pytorch, otherwise very little fastai specific knowledge needed. This notebook defines how to create the awd-lstm model. Some reading to understand decisions may be required. 

Documents progress(only one section):

- [ ] Basic NLP modules

### PR DESCRIPTION
**Pull request by arampacha**
closes #22

### CODE REVIEW

**Comment by manisnesan** [comment]
@arampacha - Very nicely done. Made a style suggestion.

File: fastai/text/models/awdlstm.py [hunk 1]
```diff
@@ -25,7 +30,12 @@ def forward(self, x):
 class WeightDropout(Module):
     "A module that wraps another layer in which some weights will be replaced by 0 during training."
 
-    def __init__(self, module, weight_p, layer_names='weight_hh_l0'):
+    def __init__(
+        self,
+        module:nn.Module, # wrapped module
+        weight_p:float, # weight dropout probability
+        layer_names:(str,Sequence[str])='weight_hh_l0' # names of the parameters to apply dropout to
+    ):
```
Comment (warner-benjamin) [inline fastai/text/models/awdlstm.py:38]: Per the recent update to the [style guide](https://github.com/muellerzr/fastai-docment-sprint/blob/master/docs_src/style-guide.md), could you remove all the newlines from the self and cls args like the example below?

And also capitalize the first word of all the docment comments? Thanks.


```suggestion
    def __init__(self,
        module:nn.Module, # Wrapped module
        weight_p:float, # Weight dropout probability
        layer_names:(str,Sequence[str])='weight_hh_l0' # Names of the parameters to apply dropout to
    ):
```

**Review by manisnesan — APPROVED** [review]

File: fastai/text/models/awdlstm.py [hunk 2]
```diff
@@ -135,7 +157,7 @@ def reset(self):
         self.hidden = [self._one_hidden(l) for l in range(self.n_layers)]
 
 # Cell
-def awd_lstm_lm_split(model):
+def awd_lstm_lm_split(model:nn.Module) -> Generator: # Generator over parameter groups
```
Comment (muellerzr) [inline fastai/text/models/awdlstm.py:160]:
```suggestion
def awd_lstm_lm_split(model:nn.Module):
```

File: fastai/text/models/awdlstm.py [hunk 3]
```diff
@@ -146,7 +168,7 @@ def awd_lstm_lm_split(model):
                           hidden_p=0.15, input_p=0.25, embed_p=0.02, weight_p=0.2, tie_weights=True, out_bias=True)
 
 # Cell
-def awd_lstm_clas_split(model):
+def awd_lstm_clas_split(model:nn.Module) -> Generator: # Generator over parameter groups
```
Comment (muellerzr) [inline fastai/text/models/awdlstm.py:171]:
```suggestion
def awd_lstm_clas_split(model:nn.Module):
```

**Review by muellerzr — CHANGES_REQUESTED** [review]
Just a few suggestions!

File: fastai/text/models/awdlstm.py [hunk 4]
```diff
@@ -63,7 +72,10 @@ def _do_nothing(self): pass
 class EmbeddingDropout(Module):
     "Apply dropout with probability `embed_p` to an embedding layer `emb`."
 
-    def __init__(self, emb, embed_p):
+    def __init__(self,
+        emb:nn.Embedding, # Wrapped embedding layer
+        embed_p:float # Embdedding <word> dropout probability
```
[the remark this answers is not in the document]
Comment (arampacha) [reply fastai/text/models/awdlstm.py:77]: Yeah, it would be nice to make it clear that specific input ids embedding vectors are zeroed entirely rather then individual embedding vector elements. So I thought that would be intuitive: like we apply dropout to "words" in vocab. Guess both phrasing are not quite perfect

**Review by warner-benjamin — APPROVED** [review]
Pending resolution of Zach's comment on:
> embed_p:float # Embedding layer dropout probability

this looks good. Thanks.
